### PR TITLE
Feature Theme App

### LINKED ISSUE #3: Update Theme and FontFamily 💅
- Colors Day/Night Mode
- FontFamily
- Generate Logo App

Theme Inspiration (https://dribbble.com/shots/19835645-Stocks-Page-UI-Trinkerr-FinTech)

### PR DESCRIPTION
**Pull request by LucasPrioste92**
- Added Icon
- New font Ubuntu
- Added new Colors 
- Update Dark/Light Theme

### CODE REVIEW

File: app/src/main/java/dev/pinkroom/marketsight/ui/theme/Color.kt [hunk 1]
```diff
@@ -2,10 +2,12 @@ package dev.pinkroom.marketsight.ui.theme
...
-val Purple80 = Color(0xFFD0BCFF)
-val PurpleGrey80 = Color(0xFFCCC2DC)
-val Pink80 = Color(0xFFEFB8C8)
-
-val Purple40 = Color(0xFF6650a4)
-val PurpleGrey40 = Color(0xFF625b71)
-val Pink40 = Color(0xFF7D5260)
\ No newline at end of file
+val Black = Color(0xFF000000)
+val BlackBackground = Color(0xFF18181B)
```
Comment (bffcorreia) [inline app/src/main/java/dev/pinkroom/marketsight/ui/theme/Color.kt]: Not an issue and not a problem at all in this project, but for future reference we are always trying in new projects to be as consistent as possible with naming. And things like `BlackBackground` are not color names :p

You can use sites like this one to find color names: https://chir.ag/projects/name-that-color

In an ideal world the designer has a scale of colors like the image below that helps, but I know it's rare :p

Screenshot 2024-04-04 at 20 13 48

File: app/src/main/java/dev/pinkroom/marketsight/ui/theme/Theme.kt [hunk 2]
```diff
@@ -16,40 +16,33 @@ import androidx.compose.ui.platform.LocalView
...
+    onSurfaceVariant = Black,
 )
 
 @Composable
 fun MarketSightTheme(
     darkTheme: Boolean = isSystemInDarkTheme(),
     // Dynamic color is available on Android 12+
-    dynamicColor: Boolean = true,
+    dynamicColor: Boolean = false,
     content: @Composable () -> Unit
```
Comment (bffcorreia) [inline app/src/main/java/dev/pinkroom/marketsight/ui/theme/Theme.kt]: I know that this code wasn't touched but always try to use a trailing comma as defined in Kotlin conventions: https://kotlinlang.org/docs/coding-conventions.html#trailing-commas

File: app/src/main/java/dev/pinkroom/marketsight/ui/theme/Type.kt [hunk 3]
```diff
@@ -1,34 +1,43 @@
...
+    titleMedium = defaultTypography.titleMedium.copy(fontFamily = FontFamilyUbuntu),
+    titleSmall = defaultTypography.titleSmall.copy(fontFamily = FontFamilyUbuntu),
+
+    bodyLarge = defaultTypography.bodyLarge.copy(fontFamily = FontFamilyUbuntu),
+    bodyMedium = defaultTypography.bodyMedium.copy(fontFamily = FontFamilyUbuntu),
+    bodySmall = defaultTypography.bodySmall.copy(fontFamily = FontFamilyUbuntu),
+
+    labelLarge = defaultTypography.labelLarge.copy(fontFamily = FontFamilyUbuntu),
+    labelMedium = defaultTypography.labelMedium.copy(fontFamily = FontFamilyUbuntu),
+    labelSmall = defaultTypography.labelSmall.copy(fontFamily = FontFamilyUbuntu)
```
Comment (bffcorreia) [inline app/src/main/java/dev/pinkroom/marketsight/ui/theme/Type.kt]: Not sure if the idea is to have specific dimensions for every typography style, but if so, we can already create our Dimens.kt file and add some default values to add in here.

Typically our file looks like this:
```kotlin
class Dimensions(
    // Padding
    val xSmallPadding: Dp = 4.dp,
    val smallPadding: Dp = 8.dp,
    val normalPadding: Dp = 16.dp,
    val largePadding: Dp = 24.dp,
    val xLargePadding: Dp = 32.dp,
    // Text
    val xSmallText: TextUnit = 12.sp,
    val smallText: TextUnit = 14.sp,
    val normalText: TextUnit = 16.sp,
    val largeText: TextUnit = 24.sp,
    val xlargeText: TextUnit = 32.sp,    
    // Etc...
)

val dimens: Dimensions
    @Composable get() {
        val configuration = LocalConfiguration.current
        return when {
            configuration.screenHeightDp <= 700 -> smallDimensions
            configuration.screenHeightDp <= 960 -> normalDimensions
            else -> largeDimensions
        }
    }

// Here you will override the dimensions needed depending on the screen size
private val smallDimensions = Dimensions()
private val normalDimensions = Dimensions()
private val largeDimensions = Dimensions()
```
You will need to adjust the file above depending on your app design guidelines.

**Review by bffcorreia — APPROVED** [review]
@LucasPrioste92 please check my comments before moving forward but it's already approved.

File: app/src/main/java/dev/pinkroom/marketsight/ui/theme/Color.kt (hunk 1 above)
Comment (LucasPrioste92) [reply app/src/main/java/dev/pinkroom/marketsight/ui/theme/Color.kt]: Thanks for the tip, I didn't know there were websites for naming colors

File: app/src/main/java/dev/pinkroom/marketsight/ui/theme/Theme.kt (hunk 2 above)
Comment (LucasPrioste92) [reply app/src/main/java/dev/pinkroom/marketsight/ui/theme/Theme.kt]: Thanks, I will change

File: app/src/main/java/dev/pinkroom/marketsight/ui/theme/Type.kt (hunk 3 above)
Comment (LucasPrioste92) [reply app/src/main/java/dev/pinkroom/marketsight/ui/theme/Type.kt]: I know the code seems strange, but the objective is just to override the theme's font family. In material3, in the Typography object, the defaultFontFamily parameter no longer exists, so the FontFamily must be changed in all Typography members. And the only way to change it is this way. https://issuetracker.google.com/issues/236358112?pli=1

**Pull request merged**